### ISSUE TITLE
Experimental support for oneAPI 2024.2

### ISSUE THREAD

**Issue by fwyzard** [alpaka-group/alpaka#2370]
The current oneAPI release is 2024.2, based on clang 19.

**Comment by fwyzard** [comment]
Due to the upstream `clang` updates this introduces a _very_ large number of warnings, like
```
/home/fwyzard/src/alpaka-group/alpaka/include/alpaka/workdiv/WorkDivHelpers.hpp:229:13: warning: 'switch' missing 'default' label [-Wswitch-default]
  229 |             switch(gridBlockExtentSubDivRestrictions)
      |             ^
```

Is this something that we want to fix, by adding an empty `default:` label at the end of every switch statement ?

Or something that we could switch off ?

**Comment by AuroraPerego** [comment]
> Is this something that we want to fix, by adding an empty `default:` label at the end of every switch statement ?

well.. 😅 
```
/data/user/aperego/alpaka/include/alpaka/workdiv/WorkDivHelpers.hpp:224:4: warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]
  224 |                         default:
      |                         ^
```

> Or something that we could switch off ?

This works to silence the warning:
```cpp
#    if BOOST_COMP_CLANG
#        pragma clang diagnostic push
#        pragma clang diagnostic ignored "-Wswitch-default"
#    endif

// ...

#    if BOOST_COMP_CLANG
#        pragma clang diagnostic pop
#    endif
```

**Comment by fwyzard** [comment]
> warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]

🤦🏻

**Comment by fwyzard** [comment]
Fixed by #2368.
